compiler/tflchef/core/src/ModelChef.cpp

```diff
@@ -882,6 +889,66 @@ void ModelChef::gather_signature_defs(const ::tflchef::ModelRecipe &model_recipe
   }
 }
 
+bool ModelChef::finalize_ext_buffer(void)
+{
+  // NOTE modification of std::string object in the middle may reallocate it.
+  // we will use std::string::reserve() to prevent this.
+
+  auto align16 = [](size_t &v) {
+    while (v % 16 != 0)
+      v++;
+  };
+
+  // get total memory for flatbuffer + all buffer_data
+  size_t result_size = _flatbuffer_builder->GetSize();
+  align16(result_size);
+  for (auto &it : _buffer_data_map)
+  {
+    std::vector<uint8_t> &buffer_data = it.second;
+    result_size += buffer_data.size();
+    align16(result_size);
+  }
+  align16(result_size);
+  result_size += 16; // additional for safety
+
+  std::string result;
```

Comment:
Just curious, is there any reason to use `string` instead of `vector<uint8_t>`?
I was wondering if there might be cases where it is misrecognized as text.

Reply:
`std::string` was referenced from tflite codes
- from `void Translator::AppendBufferData(std::string& result)`, from `r2.16` branch (5bc9d26649cca274750ad3625bd93422617eed4b)
- current HEAD is different; `void Translator::AppendBufferData(absl::Cord& result)`